packages/odf/components/create-storage-system/external-systems/CreateFlashSystem/FlashSystemConnectionDetails/system-connection-details.tsx

```diff
@@ -72,130 +97,192 @@ export const FlashSystemConnectionDetails: React.FC<
+                    </Button>
+                  </Tooltip>
+                </InputGroupItem>
+              </InputGroup>
+            )}
+          />
 
-      <FormGroupController
-        name="poolname-input"
-        control={control}
-        formGroupProps={{
-          label: t('Pool name'),
-          isRequired: true,
-          fieldId: 'poolname-input',
-        }}
-        render={({ onChange, onBlur }) => (
-          <TextInput
-            id="poolname-input"
-            value={formState.poolname}
-            type="text"
-            onChange={(_event, value: string) => {
-              onChange(value);
-              setFormState('poolname', value);
+          <FormGroupController
+            name="poolname-input"
+            control={control}
+            formGroupProps={{
+              label: t('Pool name'),
+              isRequired: true,
+              fieldId: 'poolname-input',
             }}
-            onBlur={onBlur}
-            isRequired
+            render={({ value, onChange, onBlur }) => (
+              <TextInput
+                id="poolname-input"
+                value={value}
+                type="text"
+                onChange={(_event, newValue: string) => {
+                  onChange(newValue);
+                  updateFormField('poolname', newValue);
+                }}
+                onBlur={onBlur}
+                isRequired
+              />
+            )}
           />
-        )}
-      />
-      <FormGroup label={t('Volume mode')} fieldId="volume-mode-input">
-        <Select
-          onSelect={onModeSelect}
-          id="volume-mode-input"
-          selections={inverseVolumeMapper[formState.volmode]}
-          onToggle={onToggle}
-          isOpen={isOpen}
-          isDisabled={false}
-          placeholderText={Object.keys(volumeMapper)[0]}
-        >
-          {Object.keys(volumeMapper).map((mode) => (
-            <SelectOption key={mode} value={mode} />
-          ))}
-        </Select>
-      </FormGroup>
+          <FormGroup label={t('Volume mode')} fieldId="volume-mode-input">
+            <Select
+              onSelect={onModeSelect}
+              id="volume-mode-input"
+              selections={inverseVolumeMapper[formState.volmode]}
+              onToggle={onToggle}
+              isOpen={isOpen}
+              isDisabled={false}
+              placeholderText={Object.keys(volumeMapper)[0]}
+            >
+              {Object.keys(volumeMapper).map((mode) => (
+                <SelectOption key={mode} value={mode} />
+              ))}
+            </Select>
+          </FormGroup>
+          {error && (
+            <Alert variant={AlertVariant.danger} isInline title={error} />
+          )}
+          <ActionGroup>
+            <Button
+              isLoading={isLoading}
+              spinnerAriaLabel={t('Connecting')}
+              onClick={onSubmit}
+              variant={ButtonVariant.primary}
+              isDisabled={isLoading}
```

Comment:
this should be disabled when form state is invalid, right ??
`isValid` is a flag from `useFrom` hook...
```suggestion
              isDisabled={!isValid || isLoading}
```

Reply:
we are using, right ?? check above:
```
  // Set up form validation schema
  const schema = useFlashSystemSchema();
  const resolver = useYupValidationResolver(schema);

  const { control } = useForm({
    resolver,
    mode: 'onChange',
  });
  ```